File: backend/src/main/java/net/pengcook/comment/repository/CommentRepository.java
package net.pengcook.comment.repository;

import java.util.List;
import net.pengcook.comment.domain.Comment;
import org.springframework.data.jpa.repository.JpaRepository;

public interface CommentRepository extends JpaRepository<Comment, Long> {

    List<Comment> findByRecipeId(long recipeId);

    void deleteByRecipeId(long recipeId);

    void deleteByUserId(long userId);
}

File: android/app/src/main/java/net/pengcook/android/data/repository/comment/CommentRepository.kt
package net.pengcook.android.data.repository.comment

import net.pengcook.android.presentation.core.model.Comment
import net.pengcook.android.presentation.core.model.MyComment

interface CommentRepository {
    suspend fun fetchComments(recipeId: Long): Result<List<Comment>>

    suspend fun fetchMyComments(): Result<List<MyComment>>

    suspend fun postComment(
        recipeId: Long,
        message: String,
    ): Result<Unit>

    suspend fun deleteComment(commentId: Long): Result<Unit>
}
